### ISSUE TITLE
Bug : Annotation Visible outside the graph area. Overlapping Axes Left Y

### ISSUE THREAD

**Issue by Theoooooo** [grafana/grafana#14991]
Please include this information:
### What Grafana version are you using?
5.4.3

### What datasource are you using?
ElasticSearch 6.5..4

### What OS are you running grafana on?
Debian 9.5.0

### What did you do?
Create a Graph Panel and add Annotation to the panel.
I also created a another Annotations in the dashboard settings.

### What was the expected result?
The annotation is displayed in the graph

### What happened instead?
The annotation is overlapping the numbers on the left.
![image](https://user-images.githubusercontent.com/33425208/51536907-58a47a00-1e4d-11e9-8d42-3e0236a50de3.png)

If i make the panel bigger, the resizing make the annotation dissapear from the panel (as intended)
![image](https://user-images.githubusercontent.com/33425208/51536925-6954f000-1e4d-11e9-811b-5306cc5a0c9b.png)

If the panel is too small, the annotation are always overlapping

You also can't see the annotations when you hover your mouse over it when the annotations overlapps

**Comment by marefr** [comment]
Not able to reproduce this. Can you please include your dashboard json? Is annotation timestamps outside visible time range in graph? Are annotations coming from elasticsearch or native Grafana annotations?

Thanks

**Comment by Theoooooo** [comment]
The annotations are native Grafana annotations. I create an another annotation but i use the "-- Grafana --" Datasource for them.
 The annotations should not be in this specific exemple because they where outside the "Last 24 Hours" time stamp
I don't know why but all my annotations are gone. Did they are deleted when you delete your dashboard ? 
There is the complet JSON :
`{
  "annotations": {
    "list": [
      {
        "datasource": "-- Grafana --",
        "enable": true,
        "hide": false,
        "iconColor": "#1f78c1",
        "limit": 100,
        "name": "Annotations Admin",
        "showIn": 0,
        "tags": [
          "admin"
        ],
        "type": "tags"
      },
      {
        "builtIn": 1,
        "datasource": "-- Grafana --",
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "limit": 100,
        "name": "Annotations & Alerts",
        "showIn": 0,
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "gnetId": null,
  "graphTooltip": 1,
  "id": 76,
  "iteration": 1548438165462,
  "links": [
    {
      "asDropdown": true,
      "icon": "external link",
      "tags": [
        "home"
      ],
      "title": "Dashboard Home",
      "type": "dashboards"
    },
    {
      "asDropdown": true,
      "icon": "external link",
      "tags": [
        "siel37_nsa2650_dcc"
      ],
      "title": "Dashboard Sonicwall_SIEL37_NSA2650_DCC",
      "type": "dashboards"
    }
  ],
  "panels": [
    {
      "content": "<p align=\"center\"> <img src=\"https://sieil37.fr/images/SIEIL-logo-web-L510px.jpg\" style=\"height:auto;\"> </p>",
      "gridPos": {
        "h": 5,
        "w": 5,
        "x": 0,
        "y": 0
      },
      "id": 25,
      "links": [],
      "mode": "html",
      "title": "",
      "transparent": true,
      "type": "text"
    },
    {
      "content": "<div class=\"text-center dashboard-header\">\n  <span>Dashboard Gestion Sonicwall_SIEL37_NSA2650_DCC</span>\n</div>",
      "gridPos": {
        "h": 4,
        "w": 8,
        "x": 8,
        "y": 0
      },
      "id": 4,
      "links": [],
      "mode": "html",
      "title": "",
      "transparent": true,
      "type": "text"
    },
    {
      "content": "\r\n<h2> Liens Utiles </h2>\r\n\r\n<table>\r\n   <tr>\r\n       <th>Services</th>\r\n       <th>Liens</th>\r\n   </tr>\r\n\r\n   <tr>\r\n       <td>Kibana</td>\r\n       <td><a href=\"http://172.31.18.211:5601\" target=\"_blank\">Lien Kibana</a></td>\r\n   </tr>\r\n   <tr>\r\n       <td>Cerebro</td>\r\n       <td><a href=\"http://172.31.18.211:9201/#/overview?host=http:%2F%2F172.31.18.211:9200\" target=\"_blank\">Lien Cerebro</a></td>\r\n   </tr>\r\n</table>\r\n",
      "gridPos": {
        "h": 5,
        "w": 4,
        "x": 20,
        "y": 0
      },
      "id": 12,
      "links": [],
      "mode": "html",
      "title": "Liens Utiles",
      "type": "text"
    },
    {
      "collapsed": false,
      "gridPos": {
        "h": 1,
        "w": 24,
        "x": 0,
        "y": 5
      },
      "id": 6,
      "panels": [],
      "title": "Panneau Principale Sonicwall_SIEL37_NSA2650_DCC",
      "type": "row"
    },
    {
      "aliasColors": {
        "Logs": "#1f78c1"
      },
      "bars": false,
      "dashLength": 10,
      "dashes": false,
      "datasource": "Elasticsearch SIEL37_NSA2650_DCC",
      "fill": 1,
      "gridPos": {
        "h": 10,
        "w": 12,
        "x": 0,
        "y": 6
      },
      "id": 2,
      "interval": "1d",
      "legend": {
        "alignAsTable": false,
        "avg": false,
        "current": false,
        "max": false,
        "min": false,
        "show": false,
        "total": false,
        "values": false
      },
      "lines": true,
      "linewidth": 1,
      "links": [],
      "nullPointMode": "null",
      "percentage": false,
      "pointradius": 5,
      "points": true,
      "renderer": "flot",
      "seriesOverrides": [
        {
          "alias": "/.*/",
          "color": "#1f78c1"
        }
      ],
      "spaceLength": 10,
      "stack": false,
      "steppedLine": false,
      "targets": [
        {
          "alias": "Logs",
          "bucketAggs": [
            {
              "field": "@timestamp",
              "id": "2",
              "settings": {
                "interval": "auto",
                "min_doc_count": 0,
                "trimEdges": 0
              },
              "type": "date_histogram"
            }
          ],
          "metrics": [
            {
              "field": "select field",
              "id": "1",
              "type": "count"
            }
          ],
          "query": "ID_Firewall:SIEL37_NSA2650_DCC",
          "refId": "A",
          "timeField": "@timestamp"
        }
      ],
      "thresholds": [],
      "timeFrom": null,
      "timeRegions": [
        {
          "colorMode": "custom",
          "fill": true,
          "fillColor": "rgba(160, 160, 160, 0.13)",
          "from": "12:00",
          "fromDayOfWeek": 7,
          "line": false,
          "op": "time",
          "to": "12:00",
          "toDayOfWeek": 5
        }
      ],
      "timeShift": null,
      "title": "Graphique Logs Sonicwall_SIEL37_NSA2650_DCC",
      "tooltip": {
        "shared": true,
        "sort": 0,
        "value_type": "individual"
      },
      "type": "graph",
      "xaxis": {
        "buckets": null,
        "mode": "time",
        "name": null,
        "show": true,
        "values": []
      },
      "yaxes": [
        {
          "format": "locale",
          "label": "",
          "logBase": 1,
          "max": null,
          "min": null,
          "show": true
        },
        {
          "format": "short",
          "label": null,
          "logBase": 1,
          "max": null,
          "min": null,
          "show": true
        }
      ],
      "yaxis": {
        "align": false,
        "alignLevel": null
      }
    },
    {
      "cacheTimeout": null,
      "colorBackground": false,
      "colorValue": false,
      "colors": [
        "#299c46",
        "rgba(237, 129, 40, 0.89)",
        "#d44a3a"
      ],
      "datasource": "Elasticsearch SIEL37_NSA2650_DCC",
      "format": "locale",
      "gauge": {
        "maxValue": 100,
        "minValue": 0,
        "show": false,
        "thresholdLabels": false,
        "thresholdMarkers": true
      },
      "gridPos": {
        "h": 5,
        "w": 4,
        "x": 12,
        "y": 6
      },
      "id": 8,
      "interval": null,
      "links": [],
      "mappingType": 1,
      "mappingTypes": [
        {
          "name": "value to text",
          "value": 1
        },
        {
          "name": "range to text",
          "value": 2
        }
      ],
      "maxDataPoints": 100,
      "nullPointMode": "connected",
      "nullText": null,
      "postfix": "",
      "postfixFontSize": "50%",
      "prefix": "",
      "prefixFontSize": "50%",
      "rangeMaps": [
        {
          "from": "null",
          "text": "N/A",
          "to": "null"
        }
      ],
      "sparkline": {
        "fillColor": "rgba(31, 118, 189, 0.18)",
        "full": true,
        "lineColor": "rgb(31, 120, 193)",
        "show": true
      },
      "tableColumn": "",
      "targets": [
        {
          "bucketAggs": [
            {
              "field": "@timestamp",
              "id": "2",
              "settings": {
                "interval": "auto",
                "min_doc_count": 0,
                "trimEdges": 0
              },
              "type": "date_histogram"
            }
          ],
          "metrics": [
            {
              "field": "select field",
              "id": "1",
              "type": "count"
            }
          ],
          "query": "ID_Firewall:SIEL37_NSA2650_DCC",
          "refId": "A",
          "timeField": "@timestamp"
        }
      ],
      "thresholds": "",
      "title": "Logs Sonicwall_SIEL37_SIEL37_NSA2650_DCC",
      "type": "singlestat",
      "valueFontSize": "80%",
      "valueMaps": [
        {
          "op": "=",
          "text": "N/A",
          "value": "null"
        }
      ],
      "valueName": "total"
    },
    {
      "cacheTimeout": null,
      "colorBackground": false,
      "colorValue": false,
      "colors": [
        "#299c46",
        "rgba(237, 129, 40, 0.89)",
        "#d44a3a"
      ],
      "datasource": "Elasticsearch SIEL37_NSA2650_DCC",
      "format": "locale",
      "gauge": {
        "maxValue": 100,
        "minValue": 0,
        "show": false,
        "thresholdLabels": false,
        "thresholdMarkers": true
      },
      "gridPos": {
        "h": 5,
        "w": 4,
        "x": 16,
        "y": 6
      },
      "hideTimeOverride": true,
      "id": 9,
      "interval": null,
      "links": [],
      "mappingType": 1,
      "mappingTypes": [
        {
          "name": "value to text",
          "value": 1
        },
        {
          "name": "range to text",
          "value": 2
        }
      ],
      "maxDataPoints": 100,
      "nullPointMode": "connected",
      "nullText": null,
      "postfix": "",
      "postfixFontSize": "50%",
      "prefix": "",
      "prefixFontSize": "50%",
      "rangeMaps": [
        {
          "from": "null",
          "text": "N/A",
          "to": "null"
        }
      ],
      "sparkline": {
        "fillColor": "rgba(31, 118, 189, 0.18)",
        "full": true,
        "lineColor": "rgb(31, 120, 193)",
        "show": true
      },
      "tableColumn": "",
      "targets": [
        {
          "bucketAggs": [
            {
              "field": "@timestamp",
              "id": "2",
              "settings": {
                "interval": "auto",
                "min_doc_count": 0,
                "trimEdges": 0
              },
              "type": "date_histogram"
            }
          ],
          "metrics": [
            {
              "field": "select field",
              "id": "1",
              "type": "count"
            }
          ],
          "query": "ID_Firewall:SIEL37_NSA2650_DCC",
          "refId": "A",
          "timeField": "@timestamp"
        }
      ],
      "thresholds": "3000000,3500000",
      "timeFrom": "24h",
      "title": "Logs Dernières 24h Sonicwall_SIEL37_NSA2650_DCC",
      "type": "singlestat",
      "valueFontSize": "80%",
      "valueMaps": [
        {
          "op": "=",
          "text": "N/A",
          "value": "null"
        }
      ],
      "valueName": "total"
    },
    {
      "cacheTimeout": null,
      "colorBackground": false,
      "colorValue": false,
      "colors": [
        "#299c46",
        "rgba(237, 129, 40, 0.89)",
        "#d44a3a"
      ],
      "datasource": "Elasticsearch SIEL37_NSA2650_DCC",
      "format": "locale",
      "gauge": {
        "maxValue": 100,
        "minValue": 0,
        "show": false,
        "thresholdLabels": false,
        "thresholdMarkers": true
      },
      "gridPos": {
        "h": 5,
        "w": 4,
        "x": 20,
        "y": 6
      },
      "hideTimeOverride": true,
      "id": 10,
      "interval": null,
      "links": [],
      "mappingType": 1,
      "mappingTypes": [
        {
          "name": "value to text",
          "value": 1
        },
        {
          "name": "range to text",
          "value": 2
        }
      ],
      "maxDataPoints": 100,
      "nullPointMode": "connected",
      "nullText": null,
      "postfix": "",
      "postfixFontSize": "50%",
      "prefix": "",
      "prefixFontSize": "50%",
      "rangeMaps": [
        {
          "from": "null",
          "text": "N/A",
          "to": "null"
        }
      ],
      "sparkline": {
        "fillColor": "rgba(31, 118, 189, 0.18)",
        "full": true,
        "lineColor": "rgb(31, 120, 193)",
        "show": true
      },
      "tableColumn": "",
      "targets": [
        {
          "bucketAggs": [
            {
              "field": "@timestamp",
              "id": "2",
              "settings": {
                "interval": "auto",
                "min_doc_count": 0,
                "trimEdges": 0
              },
              "type": "date_histogram"
            }
          ],
          "metrics": [
            {
              "field": "select field",
              "id": "1",
              "type": "count"
            }
          ],
          "query": "ID_Firewall:SIEL37_NSA2650_DCC",
          "refId": "A",
          "timeField": "@timestamp"
        }
      ],
      "thresholds": "110000,150000",
      "timeFrom": "1h",
      "timeShift": null,
      "title": "Logs Dernières Heure Sonicwall_SIEL37_NSA2650_DCC",
      "type": "singlestat",
      "valueFontSize": "80%",
      "valueMaps": [
        {
          "op": "=",
          "text": "N/A",
          "value": "null"
        }
      ],
      "valueName": "total"
    },
    {
      "aliasColors": {},
      "bars": false,
      "dashLength": 10,
      "dashes": false,
      "datasource": "Elasticsearch SIEL37_NSA2650_DCC",
      "fill": 1,
      "gridPos": {
        "h": 5,
        "w": 4,
        "x": 16,
        "y": 11
      },
      "hideTimeOverride": true,
      "id": 16,
      "legend": {
        "avg": false,
        "current": false,
        "max": false,
        "min": false,
        "show": false,
        "total": false,
        "values": false
      },
      "lines": true,
      "linewidth": 1,
      "links": [],
      "nullPointMode": "null",
      "percentage": false,
      "pointradius": 4,
      "points": true,
      "renderer": "flot",
      "seriesOverrides": [
        {
          "alias": "/.*/",
          "color": "#1f78c1"
        }
      ],
      "spaceLength": 10,
      "stack": false,
      "steppedLine": false,
      "targets": [
        {
          "bucketAggs": [
            {
              "field": "@timestamp",
              "id": "2",
              "settings": {
                "interval": "1h",
                "min_doc_count": 0,
                "trimEdges": 0
              },
              "type": "date_histogram"
            }
          ],
          "metrics": [
            {
              "field": "select field",
              "id": "1",
              "type": "count"
            }
          ],
          "query": "ID_Firewall:SIEL37_NSA2650_DCC",
          "refId": "A",
          "timeField": "@timestamp"
        }
      ],
      "thresholds": [],
      "timeFrom": "24h",
      "timeRegions": [],
      "timeShift": null,
      "title": "Graph Logs Dernières 24h",
      "tooltip": {
        "shared": true,
        "sort": 0,
        "value_type": "individual"
      },
      "transparent": true,
      "type": "graph",
      "xaxis": {
        "buckets": null,
        "mode": "time",
        "name": null,
        "show": false,
        "values": []
      },
      "yaxes": [
        {
          "format": "locale",
          "label": null,
          "logBase": 1,
          "max": null,
          "min": null,
          "show": true
        },
        {
          "format": "short",
          "label": null,
          "logBase": 1,
          "max": null,
          "min": null,
          "show": false
        }
      ],
      "yaxis": {
        "align": false,
        "alignLevel": null
      }
    },
    {
      "aliasColors": {},
      "bars": false,
      "dashLength": 10,
      "dashes": false,
      "datasource": "Elasticsearch SIEL37_NSA2650_DCC",
      "fill": 1,
      "gridPos": {
        "h": 5,
        "w": 4,
        "x": 20,
        "y": 11
      },
      "hideTimeOverride": true,
      "id": 17,
      "legend": {
        "avg": false,
        "current": false,
        "max": false,
        "min": false,
        "show": false,
        "total": false,
        "values": false
      },
      "lines": true,
      "linewidth": 1,
      "links": [],
      "nullPointMode": "null",
      "percentage": false,
      "pointradius": 4,
      "points": true,
      "renderer": "flot",
      "seriesOverrides": [
        {
          "alias": "/.*/",
          "color": "#1f78c1"
        }
      ],
      "spaceLength": 10,
      "stack": false,
      "steppedLine": false,
      "targets": [
        {
          "bucketAggs": [
            {
              "field": "@timestamp",
              "id": "2",
              "settings": {
                "interval": "3m",
                "min_doc_count": 0,
                "trimEdges": 0
              },
              "type": "date_histogram"
            }
          ],
          "metrics": [
            {
              "field": "select field",
              "id": "1",
              "type": "count"
            }
          ],
          "query": "ID_Firewall:SIEL37_NSA2650_DCC",
          "refId": "A",
          "timeField": "@timestamp"
        }
      ],
      "thresholds": [],
      "timeFrom": "1h",
      "timeRegions": [],
      "timeShift": null,
      "title": "Graph Logs Dernières Heure",
      "tooltip": {
        "shared": true,
        "sort": 0,
        "value_type": "individual"
      },
      "transparent": true,
      "type": "graph",
      "xaxis": {
        "buckets": null,
        "mode": "time",
        "name": null,
        "show": false,
        "values": []
      },
      "yaxes": [
        {
          "format": "locale",
          "label": null,
          "logBase": 1,
          "max": null,
          "min": null,
          "show": true
        },
        {
          "format": "short",
          "label": null,
          "logBase": 1,
          "max": null,
          "min": null,
          "show": false
        }
      ],
      "yaxis": {
        "align": false,
        "alignLevel": null
      }
    },
    {
      "content": "<div class=\"text-center dashboard-header\">\n  <span>Liste des Indices Sonicwall_SIEL37_NSA2650_DCC</span>\n</div>\n<iframe id=editor height=\"400\" width=\"1030\" style=\"background-color:white\" src=\"http://172.31.18.211:9200/_cat/indices?pretty&s=index&v&index=sonicwall_siel37_nsa2650_dcc*\"></iframe>",
      "gridPos": {
        "h": 13,
        "w": 14,
        "x": 0,
        "y": 16
      },
      "id": 27,
      "links": [],
      "mode": "html",
      "title": "",
      "transparent": true,
      "type": "text"
    }
  ],
  "refresh": "5m",
  "schemaVersion": 16,
  "style": "dark",
  "tags": [
    "sonicwall",
    "gestion",
    "general",
    "siel37_nsa2650_dcc"
  ],
  "templating": {
    "list": [
      {
        "datasource": "Elasticsearch SIEL37_NSA2650_DCC",
        "filters": [],
        "hide": 0,
        "label": "",
        "name": "Filtres",
        "skipUrlSync": false,
        "type": "adhoc"
      }
    ]
  },
  "time": {
    "from": "now-30d",
    "to": "now"
  },
  "timepicker": {
    "refresh_intervals": [
      "5s",
      "10s",
      "30s",
      "1m",
      "5m",
      "15m",
      "30m",
      "1h",
      "2h",
      "1d"
    ],
    "time_options": [
      "5m",
      "15m",
      "1h",
      "6h",
      "12h",
      "24h",
      "2d",
      "7d",
      "30d"
    ]
  },
  "timezone": "",
  "title": "Dashboard Gestion Général Sonicwall_SIEL37_NSA2650_DCC",
  "uid": "6Qpvv7qMsY",
  "version": 3
}`

**Comment by marefr** [comment]
Thanks. 

> I don't know why but all my annotations are gone. Did they are deleted when you delete your dashboard ?

Yes, annotations created for certain dashboard are deleted when that dashboard is deleted. When you say all your annotations are gone it sounds like you've hit this bug #15063 maybe?

**Comment by Theoooooo** [comment]
Thanks for the reply,

I also got this bug where the annotations are not loading correctly. I had to display the annotation checkbox and disable/activate the annotation or them to display again. But this time they are just all gone. 
I reset several dashboards so this is why all my annotations are gone.

The timestamp for the entire Dashboard is "Last 30 Days" but for the panel i shown i put in the "Time Range" option 24h in "Override relative time"
